server/threads/tests/routes/delete-thread.test.ts

```diff
@@ -0,0 +1,87 @@
+import {
+  DELETED_THREAD_ID,
+  NULL_ID,
+  NULL_THREAD_NOT_FOUND,
+  RESTRICTED_THREAD_ID
+} from "test/data/threads";
+import {
+  ENSURE_LOGGED_IN_INVALID_TOKEN,
+  ENSURE_LOGGED_IN_NO_TOKEN,
+  ENSURE_THREAD_ACCESS_UNAUTHORIZED,
+} from "test/data/responses";
+import {
+  setLoggedInUser,
+  startTestServer,
+  wrapWithTransaction,
+} from "utils/test-utils";
+
+import { BOBATAN_USER_ID } from "test/data/auth";
+import { GenericResponse } from "types/rest/responses";
+import request from "supertest";
+import router from "../../routes";
+
+jest.mock("handlers/auth");
+jest.mock("server/db-pool");
+
+describe("Tests threads REST API - delete thread", () => {
+  const server = startTestServer(router);
+
+  test("should delete thread", async () => {
+    await wrapWithTransaction(async () => {
+      setLoggedInUser(BOBATAN_USER_ID);
+      const res = await request(server.app).delete(
+        `/${DELETED_THREAD_ID}`
+      );
+
+      expect(res.status).toBe(204);
+    });
+  });
+
+  test("should fail when user is unauthenticated", async () => {
+    const res = await request(server.app).delete(
+      `/${DELETED_THREAD_ID}`
+    );
+
+    expect(res.status).toBe(401);
+    expect(res.body).toEqual<GenericResponse>(ENSURE_LOGGED_IN_NO_TOKEN);
+  });
+
+  test("should fail when thread has already been deleted/does not exist", async () => {
```

Comment:
This test is failing right now: https://app.circleci.com/pipelines/github/BobaBoard/boba-backend/686/workflows/f8364f0e-c2e5-4564-8ad2-1e5e607747e5/jobs/1153

Feel free to comment it out and add it at the next pass. We probably can get this info as part of checking if the user has permissions for deleting the thread, so no point doing it now.